libs/blockchain-watcher/feature-app/src/lib/service/appService.ts

```diff
@@ -0,0 +1,59 @@
+import express, { Request, Response } from 'express'
+import { Express } from 'express'
+import * as http from 'http'
+
+export class AppService {
+  PORT = 3001
+  app: Express | null = null
+  server: http.Server | null = null
+
+  async onHealthCheck(req: Request, res: Response) {
+    res.status(204).send()
+  }
+
+  async start(): Promise<void> {
```

Comment:
These comments apply to most of the functions in `AppService`:
1. There's no `await` used, so `async` is not needed
2. Except for `stop()`, asynchronous features are not being used.

I would rewrite `start` like this:
```ts
 start(): void {
    if (!this.app) {
      return // Or maybe throw an error, depending on usage
    }

    this.server = this.app.listen(this.PORT)
    console.log(`Api service listen on :${this.PORT}`)
  }
```

I would rewrite `stop` like this:
```ts
  async stop(): Promise<void> {
      if (!this.server) {
        return
      }
      try {
        await this.server.close()
        console.log(`Api service on :${this.PORT} closed`)
      } catch (err) {
        console.error(err)
      }
  }
```

Maybe I misunderstood the intent or usage, let me know!

Reply:
I prefer always to use `async` because it's easier to refactor when we need to add some async method into the function. About the `stop()`, we can promisify the `close()` method and it will look better.

Reply:
Ah, sorry. I misunderstood your `close()` method since I haven't directly used `http.Server` in a while. I think it's good, no changes needed 👌

For start, it's not a big problem, but generally my preference is to keep code as simple as possible, for as long as possible. If we wrap something in a promise that doesn't need to be, some developer may be confused when they look at it in the future, and waste some time determining if it really needs to be implemented that way.

Reply:
The `listen()` has a callback function so the `start()` should be an async method. However, I'll refactor them into `async/await` style :)

Reply:
🙏